dask_sql/datacontainer.py

```diff
@@ -212,9 +214,11 @@ def __call__(self, *args, **kwargs):
                     column_args.append(operand)
                 else:
                     scalar_args.append(operand)
+
             df = column_args[0].to_frame()
-            for col in column_args[1:]:
-                df[col.name] = col
+            for name, col in zip(self.names, column_args):
```

Comment:
If we pass the first parameter column name to `to_frame`, we lose a layer off the resulting HLG and don't have to deal with a superfluous column:

```
            df = column_args[0].to_frame(self.names[0])
            for name, col in zip(self.names[1:], column_args[1:]):
```